src/WSRequestHandler_Sources.cpp

```diff
@@ -1313,3 +1313,314 @@ void WSRequestHandler::HandleGetSpecialSources(WSRequestHandler* req) {
+* @return {String} `filters.*.type` Filter type
+* @return {String} `filters.*.name` Filter name
+* @return {Object} `filters.*.settings` Filter settings
+*
+* @api requests
+* @name GetSourceFilters
+* @category sources
+* @since unreleased
+*/
+void WSRequestHandler::HandleGetSourceFilters(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceName")) {
+		req->SendErrorResponse("missing request parameters");
+		return;
+	}
+
+	const char* sourceName = obs_data_get_string(req->data, "sourceName");
+	OBSSourceAutoRelease source = obs_get_source_by_name(sourceName);
+	if (!source) {
+		req->SendErrorResponse("specified source doesn't exist");
+		return;
+	}
+
+	OBSDataArrayAutoRelease filters = obs_data_array_create();
+	obs_source_enum_filters(source, [](obs_source_t *parent, obs_source_t *child, void *param)
+	{
+		OBSDataAutoRelease filter = obs_data_create();
+		obs_data_set_string(filter, "type", obs_source_get_id(child));
+		obs_data_set_string(filter, "name", obs_source_get_name(child));
+		obs_data_set_obj(filter, "settings", obs_source_get_settings(child));
+		obs_data_array_push_back((obs_data_array_t*)param, filter);
+
+	}, filters);
+
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_array(response, "filters", filters);
+	req->SendOKResponse(response);
+}
+
+/**
+* Add a new filter to a source. Available source types along with their settings properties are available from `GetSourceTypesList`.
+*
+* @param {String} `sourceName` Name of the source on which the filter is added
+* @param {String} `filterName` Name of the new filter
+* @param {String} `filterType` Filter type
+* @param {Object} `filterSettings` Filter settings
+*
+* @api requests
+* @name AddFilterToSource
+* @category sources
+* @since unreleased
+*/
+void WSRequestHandler::HandleAddFilterToSource(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceName") || !req->hasField("filterName") ||
+		!req->hasField("filterType") || !req->hasField("filterSettings"))
+	{
+		req->SendErrorResponse("missing request parameters");
+		return;
+	}
+
+	const char* sourceName = obs_data_get_string(req->data, "sourceName");
+	const char* filterName = obs_data_get_string(req->data, "filterName");
+	const char* filterType = obs_data_get_string(req->data, "filterType");
+	OBSDataAutoRelease filterSettings = obs_data_get_obj(req->data, "filterSettings");
+
+	OBSSourceAutoRelease source = obs_get_source_by_name(sourceName);
+	if (!source) {
+		req->SendErrorResponse("specified source doesn't exist");
+		return;
+	}
+
+	OBSSourceAutoRelease existingFilter = obs_source_get_filter_by_name(source, filterName);
+	if (existingFilter) {
+		req->SendErrorResponse("filter name already taken");
+		return;
+	}
+
+	OBSSourceAutoRelease filter = obs_source_create_private(filterType, filterName, filterSettings);
+	if (!filter) {
+		req->SendErrorResponse("filter creation failed");
+		return;
+	}
+	if (obs_source_get_type(filter) != OBS_SOURCE_TYPE_FILTER) {
```

Comment:
Out of curiosity, what other types could it be at this point?

Reply:
@haganbmj Filters are sources, which type is one of the `OBS_SOURCE_TYPE_*` constants.

Reply:
@haganbmj 
> what other types could it be at this point?

Actually I don't know. I just want to add another security to make sure the source created is indeed a filter.

Reply:
Yeah, I saw elsewhere in this commit where filters are `OBSSourceAutoRelease`, so makes sense to take precaution.